src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Compatibility.Common.targets

```diff
@@ -33,6 +34,37 @@
       CompatibilitySuppressionFilePath="$(CompatibilitySuppressionFilePath)"
       BaselinePackageTargetPath="$(PackageValidationBaselinePath)"
       DisablePackageBaselineValidation="$(DisablePackageBaselineValidation)"
-      RoslynAssembliesPath="$(RoslynAssembliesPath)" />
+      RoslynAssembliesPath="$(RoslynAssembliesPath)"
+      ReferenceDirectories="@(PackageValidationReferenceDirectories)" />
```

Comment:
Why do we use directories instead of actual ReferencePath items?  It's possible this allows more references to be seen than we want.

Reply:
My latest change moves to use ReferencePath. I still have the fallback to go to containing directories in order to a) not brake one API that supports adding directories to search assemblies there and b) to be able to find transitive dependencies that are not part of referencepath but happen to be on the same directories as referencePath assemblies. This latter case is rare, and usually only happens when dealing with a full facade.